src/h5web/visualizations/heatmap/HeatmapVis.tsx

```diff
@@ -30,24 +36,66 @@ function HeatmapVis(props: Props): JSX.Element {
     keepAspectRatio = true,
     showGrid = false,
     showLoader = true,
+    abscissas,
+    ordinates,
   } = props;
 
   const { rows, cols } = getDims(dataArray);
   const aspectRatio = keepAspectRatio ? cols / rows : undefined; // width / height <=> cols / rows
 
+  const abscissaIndices = range(cols + 1);
+  const abscissaValues = abscissas || abscissaIndices;
+  const abscissaToIndex = getValueToIndexScale(abscissaValues);
+  const abscissaDomain = useMemo(() => {
+    return getDataDomain(abscissaValues) ?? DEFAULT_DOMAIN;
+  }, [abscissaValues]);
+
+  const ordinateIndices = range(rows + 1);
+  const ordinateValues = ordinates || ordinateIndices;
+  const ordinateToIndex = getValueToIndexScale(ordinateValues);
+  const ordinateDomain = useMemo(() => {
+    return getDataDomain(ordinateValues) ?? DEFAULT_DOMAIN;
+  }, [ordinateValues]);
+
+  if (abscissaValues.length !== cols + 1) {
+    return (
+      <p>{`Abscissas size (${
+        abscissaValues.length
+      }) does not match data length along X (${cols + 1})`}</p>
+    );
+  }
+
+  if (ordinateValues.length !== rows + 1) {
+    return (
+      <p>{`Ordinate size (${
+        ordinateValues.length
+      }) does not match data length along Y (${rows + 1})`}</p>
+    );
+  }
+
   return (
     <div className={styles.root}>
       <VisCanvas
-        abscissaConfig={{ indexDomain: [0, cols], showGrid }}
-        ordinateConfig={{ indexDomain: [0, rows], showGrid }}
+        abscissaConfig={{
+          domain: abscissaDomain,
```

Comment:
The `domain` can now be something else than indices. That means that the scales built by `getAxisScale` that map axis coordinates to positions in the ThreeJS Scene now deal with arbitrary abscissa/ordinates.